src/librustc_error_codes/error_codes/E0657.md

```diff
@@ -0,0 +1,64 @@
+The `impl Trait` can only capture lifetimes bound at the fn or impl level.
```

Comment:
This should be part of the explanation. Please put it after the failing example and instead put: "A lifetime bound on a trait implementation was captured at an incorrect place" (If you have suggestion for a better wording @Dylan-DPC ?).